### DISCUSSION TITLE
Remove Docker image health check

### DISCUSSION THREAD

**Discussion by agneevX** [AdguardTeam/AdGuardHome#5939] [General]
I initially suggested the addition of health checks to the Docker image, but as it has become very apparent with the rapid influx of new issues, that there are many edge cases that have arisen with the check.

The check also serves no real purpose... a user running an application like autoheal can put the app in a restart loop.

Further, the health check feature consumes CPU very frequently and makes excessive writes to I/O.

> https://github.com/moby/moby/issues/39102

It would be wise to remove the check and revert changes made to the Dockerfile entrypoint/cmd options.

**Comment by ainar-g** [comment]
Besides those, removing the healthcheck would allow us to remove the `ENTRYPOINT` directive, which seems to be the culprit in many `docker-compose` and `podman` issues.

I'll pin this discussion for some time, and if there aren't any objections, we'll probably remove it.

**Comment by ainar-g** [comment]
As of Edge version `v0.108.0-a.603+f7dd8325`, the healthcheck is removed.

**Comment by up2you1** [comment]
im using AGH in Portainer and tried edge, beta and latest channels and seem to be getting this:

OCI runtime exec failed: exec failed: unable to start container process: exec: "/opt/adguardhome/scripts/healthcheck.sh": stat /opt/adguardhome/scripts/healthcheck.sh: no such file or directory: unknown

<img width="975" alt="image" src="https://github.com/AdguardTeam/AdGuardHome/assets/35837260/a9876b8b-d73d-4786-8a44-0fa7dec01673">


portainer shows AGH to be unhealthy, but DNS still working ok, any harm to leave it like it is? or how do i fix?

**Reply by ainar-g** [reply]
@up2you1, please read this note in the [Release notes](https://github.com/AdguardTeam/AdGuardHome/releases/tag/v0.107.34)

> NOTE: Some Docker tools may cache `ENTRYPOINT` sections, so some users may be required to backup their configuration, stop the container, purge the old image, and reload it from scratch.

Apparently, `HEALTHCHECK` sections are also cached.  You need to backup your configuration and recreate the container from a new image.

**Reply by up2you1** [reply]
perfect, thanks!

**Reply by peytonjol** [reply]
worked. Thanks !

**Reply by Turbine1313** [reply]
Will a recrate or duplicate container will work as well ?
Because I still get that error.
Thanks

**Reply by soundneedle** [reply]
> Will a recrate or duplicate container will work as well ?
> 
> Because I still get that error.
> 
> Thanks
> 
> 

I tried duplicating and still getting the error
